src/post-worker.js

```diff
@@ -1,758 +1,638 @@
-                    Italic: styl_ptr.get_Italic(),
-                    Underline: styl_ptr.get_Underline(),
-                    StrikeOut: styl_ptr.get_StrikeOut(),
-                    ScaleX: styl_ptr.get_ScaleX(),
-                    ScaleY: styl_ptr.get_ScaleY(),
-                    Spacing: styl_ptr.get_Spacing(),
-                    Angle: styl_ptr.get_Angle(),
-                    BorderStyle: styl_ptr.get_BorderStyle(),
-                    Outline: styl_ptr.get_Outline(),
-                    Shadow: styl_ptr.get_Shadow(),
-                    Alignment: styl_ptr.get_Alignment(),
-                    MarginL: styl_ptr.get_MarginL(),
-                    MarginR: styl_ptr.get_MarginR(),
-                    MarginV: styl_ptr.get_MarginV(),
-                    Encoding: styl_ptr.get_Encoding(),
-                    treat_fontname_as_pattern: styl_ptr.get_treat_fontname_as_pattern(),
-                    Blur: styl_ptr.get_Blur(),
-                    Justify: styl_ptr.get_Justify()
-                };
-                styles.push(style);
-            }
-            postMessage({
-                target: "get-styles",
-                time: Date.now(),
-                styles: styles
-            });
-            break;
-        case 'set-style':
-            var style = message.data.style;
-            var i = message.data.index;
-            var styl_ptr = self.octObj.track.get_styles(i);
-            _applyKeys(style, styl_ptr);
-            break;
-        case 'remove-style':
-            var i = message.data.index;
-            self.octObj.removeStyle(i);
-            break;
-        case 'runBenchmark': {
-            self.runBenchmark();
-            break;
-        }
-        case 'custom': {
-            if (Module['onCustomMessage']) {
-                Module['onCustomMessage'](message);
-            } else {
-                throw 'Custom message received but worker Module.onCustomMessage not implemented.';
-            }
-            break;
+self.canvas = data => {
+  if (data.width == null) throw new Error('Invalid canvas size specified')
+  if (Module.canvas && data.boundingClientRect) {
+    Module.canvas.boundingClientRect = data.boundingClientRect
+  }
+  self.resize(data.width, data.height)
+  self.renderLoop()
+}
+
+self.video = data => {
+  if (data.currentTime != null) self.setCurrentTime(data.currentTime)
+  if (data.isPaused != null) self.setIsPaused(data.isPaused)
+  self.rate = data.rate || self.rate
+}
+
+onmessage = message => {
+  if (!calledMain && !message.data.preMain) {
+    if (!messageBuffer) {
+      messageBuffer = []
+      messageResenderTimeout = setTimeout(messageResender, 50)
+    }
+    messageBuffer.push(message)
+    return
+  }
+  if (calledMain && messageResenderTimeout) {
+    clearTimeout(messageResenderTimeout)
+    messageResender()
+  }
+  const data = message.data
+  if (self[data.target]) {
+    self[data.target](data)
+  } else {
+    switch (data.target) {
```

Comment:
You probably forgot to implement the freeTrack, setTrack and setTrackByUrl in the onmessage event

Reply:
Also init

Reply:
This was more to reduce useless/duplicate code and make it more readable, but I'll define more functions for these

Reply:
375a943ae4324bed123d8c6db7fb1ec03f785c92